core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20210924_populate_sharded_entity_tables.xml

```diff
@@ -0,0 +1,331 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="dummy" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <!-- START populate each shard in its own changeset. If we hit an error with an
+            individual shard, we can restart the liquibase migration at that point,
+            and don't have to rollback any progress up to that changeset. -->
+    <changeSet logicalFilePath="dummy" author="davidan" id="populate_shard_00_03">
+        <sql>call copyRowsToAttributeShard('00_03');</sql>
+        <rollback>DROP TABLE IF EXISTS ENTITY_ATTRIBUTE_00_03; call createEntityAttributeShard('00_03');</rollback>
```

Comment:
what happens if the rollback fails? does Liquibase just silently move on to the next changeset or does it throw a fatal exception that prevents booting?

Reply:
i think what I am worried about is that if the `copyRowsToAttributeShard` fails once, is it likely that it will fail a 2nd time and then boot will be prevented entirely? this would require us to either go in and make a manual database change to fix the data (if that's the problematic piece) or make a quick hotfix change to the migration to fix the issue

Reply:
A few points:
* note that the rollback calls `createEntityAttributeShard`, which already succeeded in a previous changeset - so it's not a guarantee that it'll work based on current db conditions, but it's some indication of confidence.
* I tried running unit tests with the liquibase `updateTestingRollback` [command](https://docs.liquibase.com/commands/community/updatetestingrollback.html). I could see that all the sharding-related changesets rolled back correctly. But unfortunately, the overall liquibase migration failed - there are some unrelated changesets that have [missing rollbacks](https://github.com/broadinstitute/rawls/blob/f83c7d2adec00e1d5759310429c36f60cfec9103/core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20210916_clone_workspace_file_transfer.xml#L27). It gave me confidence in *these* rollbacks, though
* overall good question! And I didn't put too much thought/effort into these "populate" changesets or their rollbacks, because I wasn't sure where we'd end up for a populate strategy. Once we settle on that, I think these need more attention.

Reply:
Ugh. I'll admit, I embarrassingly misread the rollback statement. I think this is good as you have it. Thanks for clarifying :)